### PR TITLE
enh(UX) : extend Snackbar Message functionality to support HTML Strings

### PR DESCRIPTION
**Pull request by Med1233**
## Description

The PR is for enabling the Snackbar component to support HTML content. This enhancement allows for more flexibility in displaying messages and notifications within the application, making it easier to present formatted or interactive content to users.

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [x] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software

## Target serie

- [ ] 21.10.x
- [ ] 22.04.x
- [ ] 22.10.x
- [ ] 23.04.x
- [ ] 23.10.x (master)

#### Community contributors & Centreon team

- [x] I have followed the **coding style guidelines** provided by Centreon
- [ ] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [x] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

**Review by bdauria — DISMISSED** [review]

File: centreon/packages/ui/src/Snackbar/index.tsx [hunk 1]
```diff
@@ -52,6 +58,15 @@ const Snackbar = React.forwardRef(
       closeSnackbar(id);
     };
 
+    const formatedMessage =
+      typeof message === 'string' ? (
+        <div
+          dangerouslySetInnerHTML={{ __html: DOMPurify.sanitize(message) }}
```
Comment (Thebarda) [inline centreon/packages/ui/src/Snackbar/index.tsx:64]: Don't use dangerouslySetInnerHTML.
But sanitize your HTML string with this option
`, { USE_PROFILES: { html: true } }` in order to return HTML

Comment (Med1233) [reply centreon/packages/ui/src/Snackbar/index.tsx]: I changed the used library to sanitize-html

File: centreon/packages/ui/src/Snackbar/index.tsx [hunk 2]
```diff
@@ -37,6 +43,7 @@ const Snackbar = React.forwardRef(
   ): JSX.Element => {
     const { classes } = useStyles();
     const { closeSnackbar } = useSnackbar();
+    const Message = useSanitizedHTML({ initialContent: message });
```
Comment (Thebarda) [inline centreon/packages/ui/src/Snackbar/index.tsx:46]:
```suggestion
    const sanitizedMessage = useSanitizedHTML({ initialContent: message });
```

File: centreon/packages/ui/src/Snackbar/index.tsx [hunk 3]
```diff
@@ -52,6 +59,9 @@ const Snackbar = React.forwardRef(
       closeSnackbar(id);
     };
 
+    const formatedMessage =
+      typeof message === 'string' ? <div>{Message}</div> : message;
```
Comment (Thebarda) [inline centreon/packages/ui/src/Snackbar/index.tsx:63]:
```suggestion
      typeof message === 'string' ? <div>{sanitizedMessage}</div> : message;
```

**Review by Thebarda — DISMISSED** [review]

File: centreon/packages/ui/src/Snackbar/index.tsx [hunk 4]
```diff
@@ -52,6 +59,9 @@ const Snackbar = React.forwardRef(
       closeSnackbar(id);
     };
 
+    const formatedMessage =
+      typeof message === 'string' ? <div>{sanitizedMessage}</div> : message;
+
```
Comment (g10) [inline centreon/packages/ui/src/Snackbar/index.tsx:64]: the stories show this component actually accepts `message: string | Array<string>`
when it's an array the formatting is not applied

do we have the use case of multiple messages in one snackbar ? cc/ @Thebarda 

if so, then we should support it and add the formatting there to (and fix this components props)

**Review by g10 — CHANGES_REQUESTED** [review]

File: centreon/packages/ui/src/Snackbar/index.tsx (hunk 4 above)
Comment (Med1233) [reply centreon/packages/ui/src/Snackbar/index.tsx:64]: For now, we have only used it for one message, but we can extend it to handle multiple messages. However, in order to achieve that, I will need to convert it from a hook to a simple function. Therefore, we can rename useSanitizedHTML to sanitizeHTML

Comment (Med1233) [reply centreon/packages/ui/src/Snackbar/index.tsx]: I've pushed new commits, you can check the new implementation

**Review by bdauria — APPROVED** [review]

File: centreon/packages/ui/src/Snackbar/index.tsx (hunk 4 above)
Comment (g10) [reply centreon/packages/ui/src/Snackbar/index.tsx]: the issue is that the props are not correct

it should be
```
export interface SnackbarProps {
  id: string | number;
  message: string | Record<string, string>; // check the story in storybook
  severity: Severity;
}
```

now, i'm not aware if this `Record<string, string>` is used anywhere else, or only in the story

but it should also be supported imo.

**Pull request merged**